### PR TITLE
native: handle pasting Lure deeplinks in chat

### PR DESCRIPTION
**Pull request by latter-bolden**
We've seen users get confused about how to properly link to a group. Dropping an off-network share link into chat is a reasonable instinct that we should support. This PR adds the ability to extract group refs from Lure deeplinks in the same way we do if you paste a copied cite ref. Handles links at the branch level (_join.tlon.io_) and after fallback redirect (_tlon.network/lure_).

I tried to add support in a way that would make it easier to add more of these paste handlers down the road, but I could use eyes on the extracted logic — @patosullivan I think you're most familiar?

Fixes TLON-2760

### CODE REVIEW

**Comment by jamesacklin** [comment]
Highly reasonable expectation. Does this paste-action work in other inputs (Gallery rich text, Notebook BigInput)?

**Comment by latter-bolden** [comment]
> Does this paste-action work in other inputs (Gallery rich text, Notebook BigInput)

Yeah, uses the same component under the hood. 

Definitely optimized for the chat case though. As with the preexisting cite pastes there, the ux is lackluster since we don't preview inserted attachment until after you save

File: packages/ui/src/components/MessageInput/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+const logger = createDevLogger('processReference', true);
+
+export async function processReferenceAndUpdateEditor({
+  editor,
+  pastedText,
+  matchRegex,
+  processMatch,
+}: {
+  editor: EditorBridge;
```
Comment (patosullivan) [inline packages/ui/src/components/MessageInput/helpers.ts:20]: We don't use tentap on web, so we should probably do something other than passing the editor bridge here

**Review by patosullivan — CHANGES_REQUESTED** [review]
Looks good overall, but this will only work for mobile, we should probably make sure that the helpers module can work for both mobile and web.

File: packages/ui/src/components/MessageInput/helpers.ts (hunk 1 above)
Comment (latter-bolden) [reply packages/ui/src/components/MessageInput/helpers.ts]: Added the ability to pass either web or native version, how's that feel? Could also see an argument for the function accepting/returning full editor content instead

File: packages/shared/src/store/lure.ts [hunk 2]
```diff
@@ -78,14 +78,14 @@ export const useLureState = create<LureState>((set, get) => ({
       },
     });
 
-    return get().fetchLure(flag, branchDomain, branchKey);
+    // return get().fetchLure(flag, branchDomain, branchKey);
```
Comment (patosullivan) [inline packages/shared/src/store/lure.ts:81]:
```suggestion
```

File: packages/shared/src/store/lure.ts [hunk 3]
```diff
@@ -112,7 +112,7 @@ export const useLureState = create<LureState>((set, get) => ({
       json: name,
     });
 
-    return get().fetchLure(flag, branchDomain, branchKey);
+    // return get().fetchLure(flag, branchDomain, branchKey);
```
Comment (patosullivan) [inline packages/shared/src/store/lure.ts:115]:
```suggestion
```

**Review by patosullivan — APPROVED** [review]
lgtm, just a couple commented out lines that can go

**Pull request merged**